BEIMA.Backend/DeviceFunctions/GetDevice.cs

```diff
@@ -22,40 +36,48 @@ public static class GetDevice
         /// <param name="log">The logger to log to.</param>
         /// <returns>An http response containing the device information.</returns>
         [FunctionName("GetDevice")]
-        public static IActionResult Run(
+        public async Task<IActionResult> Run(
             [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "device/{id}")] HttpRequest req,
             string id,
             ILogger log)
         {
             log.LogInformation("C# HTTP trigger function processed a device get request.");
+            
+            // Check if the id is valid.
+            if (string.IsNullOrEmpty(id) || !ObjectId.TryParse(id, out _))
+            {
+                return new BadRequestObjectResult(Resources.InvalidIdMessage);
+            }
+
+            // Retrieve the device from the database.
+            ObjectId oid = new ObjectId(id);
+            BsonDocument deviceDocument = MongoDefinition.MongoInstance.GetDevice(oid);
+
+            // Check that the device returned is not null.
+            if (deviceDocument is null)
+            {
+                return new NotFoundObjectResult(Resources.DeviceNotFoundMessage);
+            }
+
+            // Get device object.
+            var device = BsonSerializer.Deserialize<Device>(deviceDocument);
 
-            // Process as device GET request for retrieving device information.
-            if (string.Equals(req.Method, "get", StringComparison.OrdinalIgnoreCase))
+            // Add url to every photo
+            foreach(var photo in device.Photos)
             {
-                // Check if the id is valid.
-                if (string.IsNullOrEmpty(id) || !ObjectId.TryParse(id, out _))
-                {
-                    return new BadRequestObjectResult(Resources.InvalidIdMessage);
-                }
-
-                // Retrieve the device from the database.
-                ObjectId oid = new ObjectId(id);
-                BsonDocument doc = MongoDefinition.MongoInstance.GetDevice(oid);
-
-                // Check that the device returned is not null.
-                if (doc is null)
-                {
-                    return new NotFoundObjectResult(Resources.DeviceNotFoundMessage);
-                }
-
-                // Return the device.
-                var dotNetObj = BsonTypeMapper.MapToDotNetValue(doc);
-                return new OkObjectResult(dotNetObj);
+                var presignedUrl = await _storage.GetPresignedURL(photo.FileUid);
+                photo.Url = presignedUrl;
```

Comment:
So these aren't automatically populated on deserialization?

Reply:
No since the urls will expire, we never store them.